lib/client.ts

```diff
@@ -0,0 +1,42 @@
+import { Kafka } from 'kafkajs';
+
+export const produceAndConsumeMessage = async function (
+  broker: string,
+  credentials: { clientID: string; clientSecret: string },
+  topic: string
+) {
+  const kafka = new Kafka({
+    brokers: [broker],
+    // authenticationTimeout: 10000,
+    // reauthenticationThreshold: 10000,
+    ssl: true,
+    sasl: {
+      mechanism: 'plain', // scram-sha-256 or scram-sha-512
+      username: credentials.clientID,
+      password: credentials.clientSecret
+    }
+  });
+
+  const producer = kafka.producer();
+
+  await producer.connect();
+  await producer.send({
+    topic: topic,
+    messages: [{ value: 'first_message' }]
+  });
+
+  await producer.disconnect();
+
+  const consumer = kafka.consumer({ groupId: 'test-group' });
+
+  await consumer.connect();
+  await consumer.subscribe({ topic: topic, fromBeginning: true });
+
+  await consumer.run({
+    eachMessage: async ({ message }) => {
+      console.log({
```

Comment:
@asanzgom I would like to review this PR only after this point has been addressed, I convert it to Draft , feel free to mark it as "Ready" as soon as you are ready for the next round of review 👍

Reply:
@andreaTP ok I will add an assertion here

Reply:
added `await expect(message.value.toString()).toContain(messageContent);`

Reply:
I don't think this actually resolves it. Consumer return resolved promise when it is attached to kafka, test code will continue right after this and it won't care about some check in async calls of `eachMessage`. Please, double check this to be sure that this check is not useless.

Reply:
@Frawless I checked it in the logs and it has to pass that check, otherwise it prints a log at ERROR level. Also I think it is out of the scope of this test and we need to speed up the test delivery. Please @andreaTP confirm

Reply:
Can I resolve this conversation? @Frawless @andreaTP